### PR TITLE
Refactor entry point of Android API, implement `requestPermissions`

### PR DESCRIPTION
**Pull request by JeroenWeener**
Before, the entry point to using the native Android API was by registering callbacks in `Android.register()`. In this new approach, a class can implement `ActivityAware` or make use of `ActivityManager`.

## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I rebased onto `next`.
- [x] I added new tests to check the change I am making, or this PR does not need tests.
- [x] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

[Contributor Guide]: https://github.com/Baseflow/flutter-permission-handler/blob/master/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: permission_handler_android/lib/src/permission_handler_android.dart [hunk 1]
```diff
@@ -42,23 +17,27 @@ class PermissionHandlerAndroid extends PermissionHandlerPlatform {
     );
   }
 
+  /// Determine whether you have been granted a particular permission.
```
Comment (mvanbeusekom) [inline permission_handler_android/lib/src/permission_handler_android.dart:20]: Use a passive voice in the API documentation:
```suggestion
  /// Determine whether the application has been granted a particular permission.
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
I really like the current setup, just have some minor remarks. Nice work

File: permission_handler_android/lib/src/android_object_mirrors/activity.dart [hunk 2]
```diff
@@ -15,3 +22,152 @@ class Activity extends JavaObject {
...
+  /// We are assuming there is only ever 1 instance of [AndroidActivity] that
+  /// is waiting for permission request results.
+  static AndroidActivity? _requestingAndroidActivity;
+
+  /// Initializes communication with native platform.
+  static void register() {
+    ActivityFlutterApi.setup(_flutterApi);
+  }
+
+  /// Gets whether you should show UI with rationale before requesting a permission.
```
[the remark this answers is not in the document]
Comment (JeroenWeener) [reply permission_handler_android/lib/src/android_object_mirrors/activity.dart:29]: This is a copy from the Android documentation. I added the link.

Do you feel like we should edit the documentation to adhere to Dart documentation standards?

File: permission_handler_android/lib/src/android_object_mirrors/activity.dart [hunk 3]
```diff
@@ -10,8 +15,44 @@ class Activity extends JavaObject {
...
+  Future<bool> shouldShowRequestPermissionRationale(
+    String permission,
+  ) {
+    return _hostApi.shouldShowRequestPermissionRationaleFromInstance(
+      this,
+      permission,
+    );
+  }
+
+  /// Determine whether you have been granted a particular permission.
```
[the remark this answers is not in the document]
Comment (JeroenWeener) [reply permission_handler_android/lib/src/android_object_mirrors/activity.dart:39]: See above comment.

File: permission_handler_android/lib/src/android_object_mirrors/activity.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply permission_handler_android/lib/src/android_object_mirrors/activity.dart]: Yes, I like the Dart version better. It'll also be more consistent as a whole.

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Pull request merged**